freqtrade/freqai/data_kitchen.py

```diff
@@ -0,0 +1,1257 @@
+            self.full_DI_values = np.append(self.full_DI_values, self.DI_values)
+        self.full_target_mean = np.append(self.full_target_mean, target_mean)
+        self.full_target_std = np.append(self.full_target_std, target_std)
+
+        return
+
+    def fill_predictions(self, len_dataframe):
+        """
+        Back fill values to before the backtesting range so that the dataframe matches size
+        when it goes back to the strategy. These rows are not included in the backtest.
+        """
+
+        filler = np.zeros(len_dataframe - len(self.full_predictions))  # startup_candle_count
+        self.full_predictions = np.append(filler, self.full_predictions)
+        self.full_do_predict = np.append(filler, self.full_do_predict)
+        if self.freqai_config.get('feature_parameters', {}).get('DI_threshold', 0) > 0:
+            self.full_DI_values = np.append(filler, self.full_DI_values)
+        self.full_target_mean = np.append(filler, self.full_target_mean)
+        self.full_target_std = np.append(filler, self.full_target_std)
+
+        return
+
+    def create_fulltimerange(self, backtest_tr: str, backtest_period: int) -> str:
+        backtest_timerange = TimeRange.parse_timerange(backtest_tr)
+
+        if backtest_timerange.stopts == 0:
+            backtest_timerange.stopts = int(datetime.datetime.now(
+                                            tz=datetime.timezone.utc).timestamp())
+
+        backtest_timerange.startts = backtest_timerange.startts - backtest_period * SECONDS_IN_DAY
+        start = datetime.datetime.utcfromtimestamp(backtest_timerange.startts)
+        stop = datetime.datetime.utcfromtimestamp(backtest_timerange.stopts)
+        full_timerange = start.strftime("%Y%m%d") + "-" + stop.strftime("%Y%m%d")
+
+        self.full_path = Path(
+            self.config["user_data_dir"]
+            / "models"
+            / str(self.freqai_config.get("identifier"))
+        )
+
+        config_path = Path(self.config["config_files"][0])
+
+        if not self.full_path.is_dir():
+            self.full_path.mkdir(parents=True, exist_ok=True)
+            shutil.copy(
+                config_path.resolve(),
+                Path(self.full_path / config_path.parts[-1]),
+            )
+
+        return full_timerange
+
+    def check_if_model_expired(self, trained_timestamp: int) -> bool:
+        """
+        A model age checker to determine if the model is trustworthy based on user defined
+        `expiration_hours` in the configuration file.
+        :params:
+        trained_timestamp: int = The time of training for the most recent model.
+        :returns:
+        bool = If the model is expired or not.
+        """
+        time = datetime.datetime.now(tz=datetime.timezone.utc).timestamp()
+        elapsed_time = (time - trained_timestamp) / 3600  # hours
+        max_time = self.freqai_config.get('expiration_hours', 0)
+        return elapsed_time > max_time
```

Comment:
I think the default here is wrong. 
if it's not set - it'll default to 0 - expiring the model immediately.

it's also kindof the only parameter that's in hours - all others are in days - so configuration is kindof inconsistent (although i'd also not want to configure this in days ...).

Reply:
Ya I think the default value is incorrect here. Thanks for noticing that!

Re: hours
This is toward the new more descriptive naming scheme that all config parameters will take. It will include units :-) so it will be train_period_days etc. I will convert all in one fell swoop when I get a chance (as the todo list indicates). 

If you have any recommendations for renaming config parameters, feel free to drop them here. In my opinion most will be getting new more descriptive names.